lib/lint.ml

```diff
@@ -77,8 +77,13 @@ module Lint = struct
 
   let run { master } job { Key.src; packages } () =
     Current.Job.start job ~pool ~level:Current.Level.Harmless >>= fun () ->
-    Current_git.with_checkout ~job src @@ fun dir ->
-    Check.of_dir ~master ~job ~packages dir
+    match List.length packages with
+    | 0 ->
+      Current.Job.log job "No packages to lint.";
+      Lwt.return (Ok ())
+    | _ ->
+      Current_git.with_checkout ~job src @@ fun dir ->
+      Check.of_dir ~master ~job ~packages dir
```

Comment:
`List.length` needs to count all the elements of a list. Emptiness are checks are therefore better done by pattern matching, which take a single operation:

```suggestion
    match packages with
    | [] ->
      Current.Job.log job "No packages to lint.";
      Lwt.return (Ok ())
    | _ ->
      Current_git.with_checkout ~job src @@ fun dir ->
      Check.of_dir ~master ~job ~packages dir
```

This is also how [`List.is_empty`](https://github.com/ocaml/ocaml/blob/d753ea1992d27aa22dcb449f4a0ac178d167d52c/stdlib/list.ml#L571-L573) works.

Reply:
Thanks for the mini-tutorial! Appreciate it. :)